packages/runtime/src/logicsig.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  Account, decodeAddress, encodeAddress,
+  generateAccount, MultiSig, MultiSigAccount,
+  multisigAddress, signBytes, verifyBytes
+} from "algosdk";
+
+import { compareArray } from "../src/lib/compare";
+import { convertToString, stringToBytes } from "../src/lib/parsing";
+
+export class LogicSig {
+  logic: Uint8Array;
+  args: Uint8Array[];
+  sig: Uint8Array;
+  msig: MultiSig | undefined;
+  SignatureAddress: string;
+
+  constructor (program: string, args: Uint8Array[]) {
+    this.logic = stringToBytes(program);
+    this.args = args;
+    this.sig = new Uint8Array(0);
+    this.msig = undefined;
+    this.SignatureAddress = generateAccount().addr;
+  }
+
+  /**
+   * Creates signature (if no msig provided) or multi signature otherwise
+   * @param secretKey sender's secret key
+   */
+  sign (account: Account, msig?: MultiSigAccount): void {
+    if (msig === undefined) {
+      this.sig = signBytes(this.logic, account.sk);
+    } else {
+      const subsigs = msig.addrs.map(addr => {
+        return { pk: decodeAddress(addr).publicKey, s: new Uint8Array(0) };
+      });
+      this.msig = {
+        v: msig.version,
+        thr: msig.threshold,
+        subsig: subsigs
+      };
+
+      const [sig, index] = this.singleSignMultisig(account, this.msig);
+      this.msig.subsig[index].s = sig;
+    }
+  }
+
+  /**
+   * Sign Multisignature
+   * @param account Account
+   * @param msig Multisignature
+   */
+  singleSignMultisig (account: Account, msig: MultiSig): [Uint8Array, number] {
+    let index = -1;
+    const passedPk = decodeAddress(account.addr).publicKey;
+    for (let i = 0; i < msig.subsig.length; i++) {
+      const pk = msig.subsig[i].pk;
+      if (compareArray(pk, passedPk)) {
+        index = i;
+        break;
+      }
+    }
+    if (index === -1) {
+      throw new Error("invalid secret key");
```

Comment:
Can we update to invalid secret key for account: `${account.addr}`?

Reply:
ok